bin/fill_block_range.rs

```diff
@@ -0,0 +1,126 @@
+    pub header_hashes: Vec<[u8; 32]>,
+    pub data_root_commitments: Vec<[u8; 32]>,
+    pub state_root_commitments: Vec<[u8; 32]>,
+    pub end_authority_set_id: u64,
+    pub end_authority_set_hash: [u8; 32],
+}
+
+async fn get_block_range_data(start_block: u32, end_block: u32) -> BlockRangeData {
+    let mut input_data_fetcher = RpcDataFetcher::new().await;
+
+    let mut start_blocks = Vec::new();
+    let mut end_blocks = Vec::new();
+    let mut header_hashes = Vec::new();
+    let mut data_root_commitments = Vec::new();
+    let mut state_root_commitments = Vec::new();
+
+    for i in (start_block..end_block).step_by(256) {
+        let block_range_end = min(i + 256, end_block);
+        let header = input_data_fetcher.get_header(block_range_end).await;
+        let (state_root_commitment, data_root_commitment) = input_data_fetcher
+            .get_merkle_root_commitments(i, block_range_end)
+            .await;
+        start_blocks.push(i);
+        end_blocks.push(block_range_end);
+        header_hashes.push(header.hash().as_bytes().try_into().unwrap());
+        data_root_commitments.push(data_root_commitment.try_into().unwrap());
+        state_root_commitments.push(state_root_commitment.try_into().unwrap());
+    }
+    let end_authority_set_id = input_data_fetcher.get_authority_set_id(end_block).await;
+    let end_authority_set_hash = input_data_fetcher
+        .compute_authority_set_hash(end_block)
+        .await;
+    BlockRangeData {
+        start_blocks,
+        end_blocks,
+        header_hashes,
+        data_root_commitments,
+        state_root_commitments,
+        end_authority_set_id,
+        end_authority_set_hash: end_authority_set_hash.0,
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    dotenv::dotenv().ok();
+    let args = FillBlockRangeArgs::parse();
+
+    let end_block = args.end_block;
+
+    let contract_address = env::var("CONTRACT_ADDRESS").expect("CONTRACT_ADDRESS must be set");
+
+    let address = contract_address
+        .parse::<Address>()
+        .expect("invalid address");
+
+    let ethereum_rpc_url = env::var("ETHEREUM_RPC_URL").expect("ETHEREUM_RPC_URL must be set");
+
+    let private_key =
+        env::var("PRIVATE_KEY").unwrap_or(String::from("0x00000000000000000000000000000000"));
+    let chain_id = env::var("CHAIN_ID").expect("CHAIN_ID must be set");
+    let wallet = LocalWallet::from_str(&private_key).expect("invalid private key");
+    let wallet = wallet.with_chain_id(chain_id.parse::<u64>().unwrap());
+
+    let provider =
+        Provider::<Http>::try_from(ethereum_rpc_url).expect("could not connect to client");
+    let client = Arc::new(SignerMiddleware::new(provider.clone(), wallet.clone()));
+
+    let contract = VectorX::new(address.0 .0, client);
+
+    let latest_block = contract.latest_block().await.unwrap();
+
+    let block_range_data = get_block_range_data(latest_block, end_block).await;
+
+    let tx: Option<TransactionReceipt> = contract
```

Comment:
Shouldn't this only be run by governance which is going to be a multisig? I think that we should just print out the calldata here and be able to paste this into a gnosis safe UI

Reply:
Adding logging with `post` option.